frontend/src/components/activePage/TableWorkspace/TableWorkspace.tsx

```diff
@@ -8,11 +8,34 @@ import TableWorkspaceRow from './TableWorkspaceRow';
 export interface ITableWorkspaceProps {
   workspaces: Array<Workspace>;
   user: User;
+  collapseAll: boolean;
+  expandAll: boolean;
+  setCollapseAll: Dispatch<SetStateAction<boolean>>;
+  setExpandAll: Dispatch<SetStateAction<boolean>>;
 }
 
 const TableWorkspace: FC<ITableWorkspaceProps> = ({ ...props }) => {
-  const { workspaces, user } = props;
-  const [expandedId, setExpandedId] = useState(['']);
+  const {
+    workspaces,
+    user,
+    collapseAll,
+    expandAll,
+    setCollapseAll,
+    setExpandAll,
+  } = props;
+  const [expandedId, setExpandedId] = useState(
+    window.sessionStorage
+      .getItem('prevExpandedIdActivePageWorkspace')
+      ?.split(',') ?? ['']
+  );
+
+  const expandWorkspace = () => {
+    setExpandedId(workspaces.map(ws => ws.id));
+  };
+
+  const collapseWorkspace = () => {
+    setExpandedId(['']);
```

Comment:
I'm probably missing something. Why not this?
```suggestion
    setExpandedId([]);
```

Reply:
i don't know why until now i had to use that notation. but now i fix it and remove ''